tensorflow_estimator/python/estimator/estimator.py

```diff
@@ -1129,13 +1186,7 @@ def _call_input_fn(self, input_fn, mode):
       ValueError: if `input_fn` takes invalid arguments.
     """
     input_fn_args = function_utils.fn_args(input_fn)
-    kwargs = {}
-    if 'mode' in input_fn_args:
-      kwargs['mode'] = mode
-    if 'params' in input_fn_args:
-      kwargs['params'] = self.params
-    if 'config' in input_fn_args:
-      kwargs['config'] = self.config
+    kwargs = self._wrap_optional_args_for_input_fn(input_fn_args, mode)
     with ops.device('/cpu:0'):
       return input_fn(**kwargs)
 
```

Comment:
There are no tests here. Please make sure that all of this functionality is unit tested, and integration tested for the distribution path.

Reply:
Thanks for comments. I have changed my coding style. The integration test for distribution path has been updated in `tensorflow` repo. https://github.com/tensorflow/tensorflow/pull/25664

Reply:
This change looks fine to me now, thanks, but we should think through how the testing of this will have to go in-- presumably that test fails without this change? So we have to push this, then that. Can we at least add a unit test here locally that checks the logic of passing the input context through the input fn?

Reply:
OK. I understand what you mean. I will add one unit test to check the `DistributionStrategy` pass but not really calling that logic.